featurebyte/schema/table.py

```diff
@@ -197,3 +217,38 @@ class ColumnSemanticUpdate(FeatureByteBaseModel):
 
     column_name: NameStr
     semantic_id: Optional[PydanticObjectId] = Field(default=None)
+
+
+class TableReadMixin(FeatureByteBaseModel):
+    """
+    Table Read Schema mixin to include datetime partition column and schema
+    """
+
+    datetime_partition_column: Optional[StrictStr] = Field(default=None)
+    datetime_partition_schema: Optional[TimestampSchema] = Field(default=None)
+
+    @model_validator(mode="before")
+    @classmethod
+    def _populate_datetime_partition_info(cls, values: Any) -> Any:
+        """
+        Populate datetime partition column and schema
+
+        Parameters
+        ----------
+        values: Any
+            Values to validate and populate
+
+        Returns
+        -------
+        Any
+        """
+        columns_info = values.get("columns_info", [])
+        for column_info in columns_info:
+            partition_metadata = column_info.get("partition_metadata")
+            if partition_metadata is not None and partition_metadata.get("is_partition_key", False):
+                dtype_metadata = column_info.get("dtype_metadata")
+                if dtype_metadata is not None:
+                    values["datetime_partition_column"] = column_info["name"]
+                    values["datetime_partition_schema"] = dtype_metadata.get("timestamp_schema")
```

Comment:
Does it mean both of these fields will only be available when retrieving the model through the API route, but not at service level? Would it be better to include these fields at the table model level so that this information is more consistently available?